app/src/main/java/org/oppia/app/home/topiclist/PromotedStoryListViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package org.oppia.app.home.topiclist
+
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModel
+import org.oppia.app.home.HomeItemViewModel
+import org.oppia.app.home.RouteToContinuePlayingListener
+import org.oppia.app.home.continueplaying.ContinuePlayingActivity
+
+/** [ViewModel] promoted story list in [HomeFragment]. */
+class PromotedStoryListViewModel(private val activity: AppCompatActivity) : HomeItemViewModel(),
+  RouteToContinuePlayingListener {
+
+  fun clickOnViewAll(@Suppress("UNUSED_PARAMETER") v: View) {
```

Comment:
I suggest removing the unused view, instead. We can call this in data-binding without passing the view:

``() -> viewModel.clickOnViewAll()``

Reply:
Updated code as per suggestion.